src/libs/actions/PaymentMethods.js

```diff
@@ -109,41 +110,36 @@ function getPaymentMethods() {
-    return DeprecatedAPI.SetWalletLinkedAccount({
-        password,
-        bankAccountID,
-        fundID,
-    })
-        .then((response) => {
-            if (response.jsonCode === 200) {
-                Onyx.merge(ONYXKEYS.USER_WALLET, {
-                    walletLinkedAccountID: bankAccountID || fundID, walletLinkedAccountType: bankAccountID ? CONST.PAYMENT_METHODS.BANK_ACCOUNT : CONST.PAYMENT_METHODS.DEBIT_CARD,
-                });
-                Growl.show(Localize.translateLocal('paymentsPage.setDefaultSuccess'), CONST.GROWL.SUCCESS, 5000);
-                return;
-            }
+function makeDefaultPaymentMethod(password, bankAccountID, fundID, previousPaymentMethodID, previousPaymentMethodType) {
+    // Optimistically set the bank account or debit card as the default payment method
+    const optimisticData = [
+        {
+            onyxMethod: 'merge',
+            key: ONYXKEYS.USER_WALLET,
+            value: {
+                walletLinkedAccountID: bankAccountID || fundID,
+                walletLinkedAccountType: bankAccountID ? CONST.PAYMENT_METHODS.BANK_ACCOUNT : CONST.PAYMENT_METHODS.DEBIT_CARD,
+            },
+        },
+    ];
 
-            // Make sure to show user more specific errors which will help support identify the problem faster.
-            switch (response.message) {
-                case CONST.WALLET.ERROR.INVALID_WALLET:
-                case CONST.WALLET.ERROR.NOT_OWNER_OF_BANK_ACCOUNT:
-                    Growl.show(`${Localize.translateLocal('paymentsPage.error.notOwnerOfBankAccount')} ${Localize.translateLocal('common.conciergeHelp')}`, CONST.GROWL.ERROR, 5000);
-                    return;
-                case CONST.WALLET.ERROR.NOT_OWNER_OF_FUND:
-                case CONST.WALLET.ERROR.INVALID_FUND:
-                    Growl.show(`${Localize.translateLocal('paymentsPage.error.notOwnerOfFund')} ${Localize.translateLocal('common.conciergeHelp')}`, CONST.GROWL.ERROR, 5000);
-                    return;
-                case CONST.WALLET.ERROR.INVALID_BANK_ACCOUNT:
-                    Growl.show(`${Localize.translateLocal('paymentsPage.error.invalidBankAccount')} ${Localize.translateLocal('common.conciergeHelp')}`, CONST.GROWL.ERROR, 5000);
-                    return;
-                default:
-                    Growl.show(Localize.translateLocal('paymentsPage.error.setDefaultFailure'), CONST.GROWL.ERROR, 5000);
-            }
-        });
+    // If this is unsuccessful, set the previous default payment method as the current default
+    const failureData = [
+        {
+            onyxMethod: 'merge',
+            key: ONYXKEYS.USER_WALLET,
+            value: {
+                walletLinkedAccountID: previousPaymentMethodID,
+                walletLinkedAccountType: previousPaymentMethodType,
+            },
+        },
+    ];
+
+    API.write('MakeDefaultPaymentMethod', {password, bankAccountID, fundID}, {optimisticData, failureData});
```

Comment:
I guess this is more of a NAB, but any reason this PR doesn't list the onyx data in the function call like https://github.com/Expensify/App/pull/9589/files?

Reply:
No particular reason but I like the idea of keeping it consistent so I've updated to have the onyx data in the function call vs as separate variables 👍🏼